src/v/cluster/health_monitor_types.h

```diff
@@ -97,11 +97,8 @@ struct partition_status
     friend bool operator==(const partition_status&, const partition_status&)
       = default;
 };
-/**
- * We keep the chunk size in partition status small to make sure that we do not
- * waste to much of memory for topics with small number of partitions.
- */
-using partition_statuses_t = ss::chunked_fifo<partition_status, 8>;
+
+using partition_statuses_t = chunked_vector<partition_status>;
```

Comment:
I actually wrote a benchmark for this change the other day here: https://github.com/redpanda-data/redpanda/pull/16656

Results are as following:

Current:

| test                                                             |  iterations |      median |         mad |         min |         max |      allocs |       tasks |        inst |
| -                                                                |           - |           - |           - |           - |           - |           - |           - |           - |
| node_health_report.serialize_many_partitions                     |         824 |     1.040ms |     3.470us |     1.036ms |     1.068ms |      66.000 |       0.000 |  23326378.6 |
| node_health_report.serialize_many_topics                         |         214 |     2.865ms |   510.776ns |     2.865ms |     2.870ms |      90.000 |       0.000 |  55992409.2 |
| node_health_report.serialize_many_topics_replicated_partitions   |         143 |     5.004ms |     6.824us |     4.995ms |     5.014ms |     158.000 |       0.000 | 101505687.9 |
| node_health_report.deserialize_many_partitions                   |         362 |     1.483ms |   903.525ns |     1.481ms |     1.484ms |    6258.000 |       0.000 |  42209025.2 |
| node_health_report.deserialize_many_topics                       |         113 |     3.552ms |   629.637ns |     3.550ms |     3.558ms |   50398.000 |       0.000 |  84158327.6 |
| node_health_report.deserialize_many_topics_replicated_partitions |          70 |     6.474ms |     3.465us |     6.468ms |     6.489ms |   50398.000 |       0.000 | 167368490.8 |

With this change:

| test                                                             |  iterations |      median |         mad |         min |         max |      allocs |       tasks |        inst |
| -                                                                |           - |           - |           - |           - |           - |           - |           - |           - |
| node_health_report.serialize_many_partitions                     |         873 |   988.066us |   621.205ns |   987.247us |   989.330us |      66.000 |       0.000 |  22272783.3 |
| node_health_report.serialize_many_topics                         |         223 |     2.811ms |     1.138us |     2.808ms |     2.819ms |      90.000 |       0.000 |  58188636.5 |
| node_health_report.serialize_many_topics_replicated_partitions   |         132 |     4.891ms |     1.530us |     4.888ms |     4.893ms |     158.000 |       0.000 | 103002728.6 |
| node_health_report.deserialize_many_partitions                   |         390 |     1.497ms |   408.038ns |     1.496ms |     1.503ms |      48.000 |       0.000 |  41676233.5 |
| node_health_report.deserialize_many_topics                       |         115 |     3.756ms |     1.158us |     3.753ms |     3.758ms |  100398.000 |       0.000 |  89721612.4 |
| node_health_report.deserialize_many_topics_replicated_partitions |          68 |     6.876ms |     2.192us |     6.871ms |     6.879ms |  100398.000 |       0.000 | 173335556.6 |

The many topics case gets slightly worse (I think mainly due to the doubled allocation count).